git/src/lib.rs

```diff
@@ -0,0 +1,206 @@
+//! A small crate providing a convienient interface to the git operations
+//! that nickel uses.
+
+use anyhow::anyhow;
+use gix::{objs::Kind, ObjectId};
+use std::{num::NonZero, path::Path};
+
+/// An error that occurred during a git operation.
+#[derive(thiserror::Error, Debug)]
+pub enum Error {
+    #[error("io error {error} at {path}")]
+    Io {
+        error: std::io::Error,
+        path: std::path::PathBuf,
+    },
+
+    #[error("target `{0}` not found")]
+    TargetNotFound(Target),
+
+    #[error(transparent)]
+    Internal(#[from] anyhow::Error),
+}
+
+pub type Result<T, E = Error> = std::result::Result<T, E>;
```

Comment:
Is this a default parameter? Meaning that if you write `Result<T>` it's equivalent as if you had defined `type Result<T> = std::result::Result<T, Error>`, but you can still use `Result<T, E'>` for `E' != Error` ? I didn't know about this!